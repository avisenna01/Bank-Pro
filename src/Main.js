import React from "react";
import Header from "./components/Header";
import LoginPage from "./views/LoginPage";
import Grid from "@material-ui/core/Grid";
import { BrowserRouter as Router, Route } from "react-router-dom";
import AutoRoute from "./AuthRoute"

const Main = () => {
  return (
    <Router>
      <Grid>
        <Header />
        <Route path="/login" component={LoginPage} />
        {/* <Route path="/" component={HomePage} /> */}

      </Grid>
    </Router>
  );
};

export default Main;
