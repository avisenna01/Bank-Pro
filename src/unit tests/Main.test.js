import React from 'react'
import Main from "../Main";
import renderer from 'react-test-renderer';
import ReactDOM from "react-dom";
import { shallow } from 'enzyme';

const rendercomponent = () => (shallow(<Main />));

describe('Main page try 1', () => {
    it('snapshot renders', () => {
        const renderedcomponent = rendercomponent();
        expect(renderedcomponent.find('div').length).toEqual(1);
    });
});

describe('Main page try 2', () => {
    test('snapshot renders', () => {
        const component = renderer.create(<Main />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Main />, div);
});