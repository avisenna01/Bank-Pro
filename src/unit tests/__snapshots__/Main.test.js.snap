// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page try 2 snapshot renders 1`] = `
<div
  className="MuiGrid-root"
  computedMatch={
    Object {
      "isExact": true,
      "params": Object {},
      "path": "/",
      "url": "/",
    }
  }
  location={
    Object {
      "hash": "",
      "pathname": "/",
      "search": "",
      "state": undefined,
    }
  }
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 makeStyles-root-104 MuiPaper-rounded"
  >
    <img
      alt="logo"
      height="24px"
      src="apple1.png"
      width="20px"
    />
  </div>
</div>
`;
